v4/source/teams.yaml

```diff
@@ -968,3 +968,58 @@
                 type: string
         '400':
           $ref: '#/responses/BadRequest'
+
+'/teams/{team_id}/schemes/{scheme_id}':
+    put:
```

Comment:
I don't think I've seen this pattern before in our APIs (but that might be just my ignorance) where we have URL parameters and an empty `PUT` to set some field on an object.

Reply:
I chose `PUT` because the request is idempotent and a zero length request body because it would be unnecessarily verbose (and isn't required by the spec). One objection to using a `PUT` [from the HTTP 1.1 spec](https://tools.ietf.org/html/rfc7231#section-4.3.4):

> A successful PUT of a given representation would suggest that a subsequent GET on that same target resource will result in an equivalent representation being sent in a 200 (OK) response.

...however it's just a suggestion and anyhow APIv4 does not adhering to that, for example:

* `/api/v4/teams/{{team_id}}/members/{{user_id}}/roles`
* `/api/v4/channels/{{channel_id}}/members/{{user_id}}/roles`
* `/api/v4/channels/{{channel_id}}/members/{{user_id}}/notify_props`

Did you have a specific argument for changing the HTTP verb or URL structure?

Reply:
I think the weird pattern, at least for me, is passing the "new data" directly in the url, I would expect something like a PUT or a POST but in the url  `/teams/{team_id}/schemes` and with the body `{scheme_id: 'XXXXXXX'}`.

Reply:
Consistency is an important part of the API so I agree with @jespino, I'd prefer having it in the request body as that's what the rest of our endpoints do.

As for `PUT` vs `POST`, I'm with @mkraft on using `PUT` as that's what we use in the rest of the API for update semantics

Reply:
Well looks like it's 3 against 1 here so I've got some typing to do.